### PR TITLE
Clarify steps for publishing a Dandiset

### LINKED ISSUE #110: Provide explicit guidance/expectations for zarr uploads
Zarr uploads still remain in `Draft` mode, as they are yet to be versioned. 

Our handbook does not provide documentation for "what to expect" for Zarr uploads in the DANDI ecosystem. (e.g. https://github.com/dandi/handbook/blob/master/docs/13_upload.md does not contain any reference)

The goal of this Issue is to capture follow-up work to communicate the Zarr workflow correctly in this handbook, `dandi-cli` and `dandi-archive`

Relates to https://github.com/dandi/dandi-archive/issues/1811

### PR DESCRIPTION
**Pull request by kabilar**
## Changes
- [x] Convert current content into a list
- [x] Refactor list to better follow publishing flow
- [x] Add a note that Dandisets with Zarr assets currently cannot be published
  - [x] Fix #110

cc @aaronkanzer

### CODE REVIEW

**Review by waxlamp — COMMENTED** [review]
I pushed a commit with some editing. @kabilar take a look and let me know what you think.

**Comment by kabilar** [comment]
> I pushed a commit with some editing. @kabilar take a look and let me know what you think.

Thank you, @waxlamp.  Those changes look good to me.

File: docs/14_publish.md [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
+   to edit the following fields:
+    - People and funding contributors
+    - Protocol information
+    - Keywords
+    - Related resources such as publications and code repositories
+
+1. Fix all asset metadata errors by modifying the asset files to eliminate
+   the errors and re-uploading them.
+
+1. When all the Dandiset metadata and asset metadata errors are fixed, the
```
Comment (yarikoptic) [inline docs/14_publish.md:26]:
```suggestion
1. When all the Dandiset metadata and asset metadata errors are fixed, and the dandiset made public if it was initially embargoed, the
```

I think worth mentioning, but we might need a dedicated doc to describe how.

Comment (yarikoptic) [reply docs/14_publish.md:26]: ping on this one @kabilar so we could potentially quickly merge this one

File: docs/14_publish.md [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
+   to edit the following fields:
+    - People and funding contributors
+    - Protocol information
+    - Keywords
+    - Related resources such as publications and code repositories
+
+1. Fix all asset metadata errors by modifying the asset files to eliminate
+   the errors and re-uploading them.
+
+1. When all the Dandiset metadata and asset metadata errors are fixed, and the dandiset made public if it was initially embargoed, the
```
Comment (kabilar) [inline docs/14_publish.md:26]:
```suggestion
1. When all the Dandiset metadata and asset metadata errors are fixed, and the Dandiset is made public if it was initially embargoed, the
```

File: docs/14_publish.md (hunk 1 above)
Comment (kabilar) [reply docs/14_publish.md]: Thanks @yarikoptic.  Just added this suggestion.

**Pull request merged**